crates/matrix-sdk/src/send_queue.rs

```diff
@@ -1136,7 +1171,7 @@ pub enum LocalEchoContent {
         send_handle: SendHandle,
         /// Whether trying to send this local echo failed in the past with an
         /// unrecoverable error (see [`SendQueueRoomError::is_recoverable`]).
-        is_wedged: bool,
+        send_error: Option<QueueWedgeError>,
```

Comment:
I might have misunderstood your intent, I thought the goal was to use a `matrix_sdk::Error` here, since it's the most generic type for signalling the error to the caller. Can we do that?

Reply:
Why not? After deserializing a `QueueWedgeError`, isn't there a way to turn it into a `matrix_sdk::Error`, by the way of conversion?

Reply:
Answering myself: no, and prior to the patch we only had the bool "is wedged or not", so this is already quite an improvement. And indeed in other contexts that we care more about, we can keep the `crate::Error`, so the current code is fine as is. Resolving this comment, thanks for the ping :)